### ISSUE TITLE
Match statement production - correct?

### ISSUE THREAD

**Issue by hrennau** [ballerina-platform/ballerina-spec#371]
**Description:**
If I read the production match-pattern correctly, the following match-pattern would be valid:
{x: "abc", y: var y}

In particular, I refer to:

match-pattern := var binding-pattern | ... | mapping-match-pattern | ...
mapping-match-pattern := { field-match-patterns }
field-match-patterns := field-match-pattern (, field-match-pattern)* [, rest-match-pattern] | ...
field-match-pattern := field-name : match-pattern

However, running the code triggers error messages:
error: .::match_experiment_01.bal:20:23: invalid token 'var'
error: .::match_experiment_01.bal:20:23: extraneous input 'var'

Version info:
Ballerina 1.0.0
Language specification 2019R3

**Comment by jclark** [comment]
@hrennau That fragment looks fine to me. Can you attach the complete file?

**Comment by hrennau** [comment]
Source code attached.
[code_issue_371.zip](https://github.com/ballerina-platform/ballerina-spec/files/3801370/code_issue_371.zip)

**Comment by jclark** [comment]
@hasithaa This looks like a compiler bug. Can you transfer the issue?

**Comment by hasithaa** [comment]
@hrennau @jclark Yes, Current implementation doesn't support this scenario. As a workaround, you can use the following. 


```ballerina
import ballerina/io;

public function main() {
    map<anydata> r = {x: "abc", y:1};
    demomatch(r);
}

public function demomatch(anydata v) {
    match v {
        var {x, y} if x == "abc" => { io:println("alter 1"); }
        var z => { io:println("Default branch"); }
    }
}
```

I will transfer this issue.
